engine/src/main/java/com/datatorrent/stram/api/OperatorDeployInfo.java

```diff
@@ -314,6 +319,10 @@ public void sendMetrics(Collection<String> metricNames)
   public AttributeMap contextAttributes;
 
   /**
+   * Operator deploy reason groupId (event group id)
+   */
+  public long deployGroupId;
```

Comment:
groupId should be unique, so I am using timestamp to generate groupIds.
GroupIds are generated,
1. At container: When operator throws an exception and reports an error to StrAM using "reportError" method, groupId is generated.
2. At Stram: When stram detects a container has been killed it  generates a groupId.

The generated groupId is then used for redeploy of remaining operators.